generator/altitude_generator.cpp

```diff
@@ -0,0 +1,86 @@
+#include "generator/routing_generator.hpp"
+#include "generator/srtm_parser.hpp"
+
+#include "defines.hpp"
+
+#include "routing/routing_helpers.hpp"
+
+#include "indexer/feature.hpp"
+#include "indexer/feature_altitude.hpp"
+#include "indexer/feature_data.hpp"
+#include "indexer/feature_processor.hpp"
+
+#include "coding/file_container.hpp"
+#include "coding/varint.hpp"
+
+#include "coding/internal/file_data.hpp"
+
+#include "base/assert.hpp"
+#include "base/logging.hpp"
+#include "base/string_utils.hpp"
+
+#include "std/map.hpp"
+#include "std/type_traits.hpp"
+
+using namespace feature;
+
+namespace
+{
+static_assert(is_same<TAltitude, generator::SrtmTile::THeight>::value, "");
+static_assert(kInvalidAltitude == generator::SrtmTile::kInvalidHeight, "");
+
+class Processor
+{
+public:
+  Processor(string const & srtmPath) : m_srtmManager(srtmPath) {}
+  map<uint32_t, Altitudes> const & GetFeatureAltitudes() const { return m_featureAltitudes; }
+
+  void operator()(FeatureType const & f, uint32_t const & id)
+  {
+    f.ParseTypes();
+    f.ParseHeader2();
+    if (!routing::IsRoad(feature::TypesHolder(f)))
+      return;
+
+    f.ParseGeometry(FeatureType::BEST_GEOMETRY);
+    size_t const pointsCount = f.GetPointsCount();
+    if (pointsCount == 0)
+      return;
+
+    Altitudes alts(m_srtmManager.GetHeight(MercatorBounds::ToLatLon(f.GetPoint(0))),
+                   m_srtmManager.GetHeight(MercatorBounds::ToLatLon(f.GetPoint(pointsCount - 1))));
+    m_featureAltitudes[id] = alts;
+  }
+
+private:
+  generator::SrtmTileManager m_srtmManager;
+  map<uint32_t, Altitudes> m_featureAltitudes;
```

Comment:
Look at the code you asked me to add. Are you sure it's better now? In my opinion it became less readable. Anyway it's not very important in the context of the PR because |m_featureAltitudes| will be removed. A better method of keeping altitudes will be implemented.